mission/TAC_tasks/scripts/pipeline_following/pipeline_following_sm.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python3
+
+import rospy
+from smach import StateMachine
+from smach_ros import IntrospectionServer
+from pipeline_following import PipelineExecute, PipelineStandby
+from task_manager_defines import defines
+import dynamic_reconfigure.client
+
+
+def task_manager_cb(config):
+    rospy.loginfo(
+        """Client: state change request: {Tac_states}""".format(**config))
+    activated_task_id = config["Tac_states"]
+
+    if defines.Tasks.valve_vertical.id == activated_task_id:
+        isEnabled = True
+    else:
+        isEnabled = False
+    print(f"isEnabled: {isEnabled} ")
+
+
+def main():
+    rospy.init_node("tac_pipeline_fsm")
+    # rospy.wait_for_message()
+    # rospy.wait_for_service()
+
+    pipeline_following_sm = StateMachine(outcomes=["done"])
+    with pipeline_following_sm:
+
+        StateMachine.add("PIPELINE_STANDBY",
+                         PipelineStandby(),
+                         transitions={
+                             "aborted": "done",
+                             "succeeded": "PIPELINE_EXECUTE"
+                         }),
+
+        StateMachine.add("PIPELINE_EXECUTE",
+                         PipelineExecute(),
+                         transitions={"aborted": "PIPELINE_STANDBY"})
+
+    #intro_server = IntrospectionServer(
+    #    str(rospy.get_name()), pipeline_following_sm, "/SM_ROOT"
+    #)
+    #intro_server.start()
+
+    try:
+        #Execute SMACH plan
+        pipeline_following_sm.execute()
+        rospy.loginfo("Exiting Pipeline Following")
+        #intro_server.stop()
+
+    except Exception as e:
+        rospy.loginfo("State machine failed: %s" % e)
+
+
+if __name__ == "__main__":
+    while not rospy.is_shutdown():
+        # task manager
+        isEnabled = False
+        task_manager_client = dynamic_reconfigure.client.Client(
+            "task_manager/task_manager_server",
+            timeout=5,
+            config_callback=task_manager_cb)
+        while not rospy.is_shutdown() and isEnabled:
+            rospy.loginfo('STARTING PIPELINE FOLLOWING')
+            main()
+            rospy.loginfo('PIPELINE FOLLOWING ENDED')
```

Comment:
Remove the nested while loop. Move task_manager init elsewhere.

Reply:
I changed the nested while loop to an if statement "if isEnabled:", but not sure what to do about the client. Should i make a class? And will that work with the smach state machine?